docs/DynamicCasting.md

```diff
@@ -334,49 +347,44 @@ Any class, enum, struct, tuple, function, metatype, or existential metatype inst
 
 (See "The _ObjectiveCBridgeable Protocol" below for more details.)
 
-### Objective-C Interactions
-
-Note the invariant above cannot be an equality because Objective-C bridging allows libraries to introduce new relationships that can alter the behavior of seemingly-unrelated casts.
-One example of this is Foundation's `NSNumber` type which conditionally bridges to several Swift numeric types.
-As a result, when Foundation is in scope, `Int(7) is Double == false` but `(Int(7) as! AnyObject) is Double == true`.
-In general, the ability to add new bridging behaviors from a single type to several distinct types implies that Swift casting cannot be transitive.
-
 ### Error (SE-0112)
 
-Although the Error protocol is specially handled by the Swift compiler and runtime (as detailed in [SE-0112](https://github.com/apple/swift-evolution/blob/master/proposals/0112-nserror-bridging.md)), it behaves like an ordinary protocol type for casting purposes.
+The Error type behaves like an ordinary existential type for casting purposes.
 
 (See "Note: 'Self-conforming' protocols" below for additional details relevant to the Error protocol.)
 
 ### AnyHashable (SE-0131)
 
 For casting purposes, `AnyHashable` behaves like an existential type.
+It satisfies the weak existential invariant above.
 
 However, note that `AnyHashable` does not act like an existential for other purposes.
 For example, it's metatype is named `AnyHashable.Type` and it does not have an existential metatype.
 
 ### Protocol Witness types
 
-Caveat:
-Protocols that have `associatedtype` properties or which make use of the `Self` typealias cannot be used as independent types.
-As such, the discussion below does not apply to them.
+Any protocol definition (except those that include an `associatedtype` property or which makes use of the `Self` typealias) has an associated existential type named after the protocol.
 
-Any Swift instance of a concrete type `T` can be cast to `P` iff `T` conforms to `P`.
-The result is a "protocol witness" instance that provides access only to those methods and properties defined on `P`.
+Specifically, assume you have a protocol definition
+```
+protocol P {}
+```
+
+As a result of this definition, there is an existential type (also called `P`).
+This existential type is also known as a "protocol witness" type since it exposes exactly the capabilities that are defined by the protocol.
```

Comment:
```suggestion
This existential type is also known as a "protocol witness type" since it exposes exactly the capabilities that are defined by the protocol.
```

Reply:
Done.